agent2: Use inline enums in `now` and `edit_file` tools JSON schema

fbo25 opened the pull request:
Added schemars annotations to generate inline enums instead of references ($ref) in the JSON schema passed to LLMs.

Concerns :
- "timezeone" parameter for "now" tool function
- "mode" parameter for "edit_file" tool function

Should be the same for futures tools/functions enums. This is easier for LLMs to understand the schema since many of them don't use JSON references correctly.

Tested with :
- local GPT-OSS-120b with llama.cpp server (openai compatible)
- remote Claude Sonnet 4.0 with Zed pro subscription

Thanks in advance for the merge.
(notice this is my first PR ever on Github, I hope I'm doing things well, please let me know if you have any comment - edit: just noticed my username/email were not correctly setup on my local git, sorry, it's been 5 years I've not used git)

Closes #37389

Release Notes:

- agent: Improve "now" and "edit_file" tool schemas to work with more models.

imumesh18 commented:
Hey @fbo25 you will have to add your email which you have used to sign the cla and force push the commit again for cla bot to work. Make sure you are on `fbo/fix-json-schema-enums` branch

```
Step 1: git config --global user.email email@example.com
Step 2: git commit --amend --no-edit
Step 3: git push origin fbo/fix-json-schema-enums -f
```

fbo25 committed:
Use inline enums in "now" and "edit_file" tools JSON schema (#37389) ->
added schemars annotations

Concerns :
- "timezeone" parameter for "now"
- "mode" parameter for "edit_file"

Should be the same for futures enums. This is easier for LLMs to
understand the schema since many of them don't use JSON references
correctly.

Tested with :
- local GPT-OSS-120b with llama.cpp server (openai compatible)
- remote Claude Sonnet 4.0 with Zed pro subscription

Thanks in advance for the merge.

fbo25 commented:
Thank you [imumesh18](https://github.com/imumesh18) !

benbrandt reviewed (approved):
Great catch! Thanks for the contribution!
